src/extensionsIntegrated/CustomSnippets/helper.js

```diff
@@ -0,0 +1,778 @@
+/*
+ * GNU AGPL-3.0 License
+ *
+ * Copyright (c) 2021 - present core.ai . All rights reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://opensource.org/licenses/AGPL-3.0.
+ *
+ */
+
+define(function (require, exports, module) {
+    const StringMatch = require("utils/StringMatch");
+    const Global = require("./global");
+    const UIHelper = require("./UIHelper");
+
+    // list of all the navigation and function keys that are allowed inside the input fields
+    const ALLOWED_NAVIGATION_KEYS = [
+        "Backspace",
+        "Delete",
+        "Tab",
+        "Escape",
+        "Enter",
+        "ArrowLeft",
+        "ArrowRight",
+        "ArrowUp",
+        "ArrowDown",
+        "Home",
+        "End",
+        "PageUp",
+        "PageDown",
+        "F1",
+        "F2",
+        "F3",
+        "F4",
+        "F5",
+        "F6",
+        "F7",
+        "F8",
+        "F9",
+        "F10",
+        "F11",
+        "F12"
+    ];
+
+    /**
+     * map the language IDs to their file extensions for snippet matching
+     * this is needed because we expect the user to enter file extensions and not the file type inside the input field
+     *
+     * @param {string} languageId - The language ID from Phoenix
+     * @returns {string} - The equivalent file extension for snippet matching
+     */
+    function mapLanguageToExtension(languageId) {
```

Comment:
use `language = LanguageManager.getLanguageForPath(editor.document.file._path);` or `languageId = LanguageManager.getLanguageForPath(editor.document.file._path).getId();` to reliably get the language. this will honor users custom file type associations as well.